### PR TITLE
add  always visible lock icon on top right to show info about end to end encryption

### PR DESCRIPTION
**Pull request by ad1992**
I have added in top right as discussed in https://github.com/excalidraw/excalidraw/issues/1313

Screenshot 2020-04-15 at 12 42 28 AM

### CODE REVIEW

**Comment by lipis** [comment]
It's too prominent.. maybe without a background, smaller and green?

File: src/components/LayerUI.tsx [hunk 1]
```diff
@@ -39,6 +40,27 @@ interface LayerUIProps {
   onLockToggle: () => void;
 }
 
+function renderEncryptedIcon() {
```
Comment (lipis) [inline src/components/LayerUI.tsx:43]: can you use an arrow function here?

File: src/components/LayerUI.tsx [hunk 2]
```diff
@@ -39,6 +40,27 @@ interface LayerUIProps {
...
+  return (
+    <div className="encrypted-icon tooltip">
+      <span className="tooltip-text">
+        {t("encrypted")}{" "}
+        <a href="https://blog.excalidraw.com/end-to-end-encryption/">
+          {"Read our blog post to learn more"}
```
Comment (lipis) [inline src/components/LayerUI.tsx:49]: add this to the `en.json`

**Comment by ad1992** [comment]
> It's too prominent.. maybe without a background, smaller and green?

I have updated as per your suggestion and attached screenshot above

**Comment by ad1992** [comment]
what icon should be then used for locking/unlocking shapes ?

**Comment by pshihn** [comment]
Some icon overlap at certain sizes. 
Screen Shot 2020-04-12 at 9 45 20 AM

**Comment by ad1992** [comment]
> Some icon overlap at certain sizes.
> Screen Shot 2020-04-12 at 9 45 20 AM

Thank you for pointing this out. This is fixed.

**Comment by lipis** [comment]
I think it's still a bit big.. and the font size of tool tip vould be smaller as well.

**Comment by lipis** [comment]
Also use the open color variables for all colors. https://github.com/excalidraw/excalidraw/pull/1371

**Comment by ad1992** [comment]
@lipis I have made the changes as per your suggestions.

**Comment by ad1992** [comment]
@lipis did you get a chance to check this?

**Comment by lipis** [comment]
It's fine by me..! :) We just need to decide if we want it or not..

/cc @vjeux

**Comment by dwelle** [comment]
How about we use a shield icon instead?

image

We can also use it elsewhere in the UI where we use lock to describe our e2e.

**Comment by lipis** [comment]
Shield sounds better! and we don't have too many locks in the UI

File: src/components/LayerUI.tsx [hunk 3]
```diff
@@ -40,6 +42,24 @@ interface LayerUIProps {
   onLockToggle: () => void;
 }
 
+const renderEncryptedIcon = () => {
+  return (
+    <div className="encrypted-icon tooltip">
+      <span className="tooltip-text">
+        {t("encrypted.prefix")}{" "}
```
Comment (lipis) [inline src/components/LayerUI.tsx]:
```suggestion
        {t("encrypted.prefix")}<br>
```

Comment (dwelle) [reply src/components/LayerUI.tsx]: huh.. I just did that right before your suggestion.. but forgot to remove the space.

Comment (dwelle) [reply src/components/LayerUI.tsx]: Actually, the space was removed by linter.

File: src/components/LayerUI.scss [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      top: 100%;
+      right: 0;
+      background-color: $oc-black;
+      color: $oc-white;
+      text-align: center;
+      border-radius: 6px;
+      padding: 5px;
+      position: absolute;
+      z-index: 10;
+      font-size: 13px;
```
Comment (lipis) [inline src/components/LayerUI.scss:32]: maybe adding a bigger `line-height`, seems tight.

File: src/components/LayerUI.tsx [hunk 5]
```diff
@@ -195,9 +215,7 @@ export const LayerUI = React.memo(
             </Stack.Col>
           </div>
         </FixedSideContainer>
-        <aside>
-          <GitHubCorner />
```
Comment (lipis) [inline src/components/LayerUI.tsx:199]: Let's also remove teh GitHubCorner completely from the codebase.

**Comment by dwelle** [comment]
Yea, I'll figure something out because I think the GH is valuable, as well.

**Comment by dwelle** [comment]
I think we have only 2 options:

- bottom left

    image

- bottom right

    image

All others looked bad or had some UX issue.

**Comment by dwelle** [comment]
> Could you switch this over to use the popover used by the color picker? It would also be good to have some vertical space between the icon and the popup.

Both would likely force us to roll-out custom eventing for show/hide otherwise user wouldn't be able to move mouse over to the tooltip without it hiding. It's awkward even now, so we'll prob wanna do it later, anyway, but for now I think it's overkill.

Feel free to push to this PR, though, if you're up for it.

**Comment by dwelle** [comment]
To actually simplify this further, how about we eschew custom tooltip completely, for now, and go with a combination of native tooltip that says `yada yada. Click for more...`, and add a `onClick` handler on the shield icon that leads to the blog post?

**Comment by ad1992** [comment]
I think tooltip will be a better option than adding an onclick handler which actually takes to the blog post because the main motive behind adding this icon is communicating to the user that the data is encrypted which looks simpler and intuitive by just hovering on the icon as compared to showing something like "click for more..." where one more extra step is needed which actually takes the user to the blog post.

**Comment by dwelle** [comment]
Actually, in both cases (current and what I proposed) you need to click somewhere. Currently you need to hover + click on the link. In my proposal you'd hover + click on shield, or click on shield directly (as it would have `pointer: cursor`).

But I agree that the `click for more...` tooltip is not good. That said, I think making the shield clickable as a bonus isn't a bad idea either way.

**Comment by ad1992** [comment]
Yes initally I also suggested bottom left so that GitHub icon is still there.  What I meant is use might or might not click on the blog post when seeing the tooltip so showing info about encryption in tooltip itself would be helpful. So yes if even we remove the link from tooltip and make the sheild clickable and keep tooltip same(just removing the blog post link) also sounds good.

**Comment by dwelle** [comment]
The 2nd commit I just pushed makes the shield into `<a>`, which then makes even to tooltip clickable as a whole, so the inner tooltip link is there just for visuals, effectively.

Not sure about this.

**Comment by dwelle** [comment]
I pushed some crazy CSS rules to improve UX around drawing.

File: src/components/LayerUI.tsx [hunk 6]
```diff
@@ -40,6 +43,29 @@ interface LayerUIProps {
   onLockToggle: () => void;
 }
 
+const renderEncryptedIcon = () => {
+  return (
```
Comment (lipis) [inline src/components/LayerUI.tsx:47]: can be simplified... no need of return

**Comment by ad1992** [comment]
Since the shield is now an anchor tag itself so can we remove the link from the tooltip? And change the text of the tooltip to something like `Your drawings are end-to-end encrypted so Excalidraw's servers will never see them. Click to learn more`?

**Comment by dwelle** [comment]
I agree with the tooltip link. Changed it as you suggested. Let's ship this, we can iterate later if need be.

Thanks for the PR!

**Pull request merged**